query/statement_rewriter.go

```diff
@@ -69,25 +69,39 @@ func rewriteShowFieldKeyCardinalityStatement(stmt *influxql.ShowFieldKeyCardinal
 
 	return &influxql.SelectStatement{
 		Fields: []*influxql.Field{
-			{
+			&influxql.Field{
 				Expr: &influxql.Call{
 					Name: "count",
 					Args: []influxql.Expr{
 						&influxql.Call{
 							Name: "distinct",
-							Args: []influxql.Expr{&influxql.VarRef{Val: "_fieldKey"}},
+							Args: []influxql.Expr{&influxql.VarRef{Val: "fieldKey"}},
 						},
 					},
 				},
 				Alias: "count",
 			},
 		},
-		Sources:    rewriteSources2(stmt.Sources, stmt.Database),
 		Condition:  stmt.Condition,
 		Dimensions: stmt.Dimensions,
-		Offset:     stmt.Offset,
-		Limit:      stmt.Limit,
 		OmitTime:   true,
+		Sources: influxql.Sources{
+			&influxql.SubQuery{
+				Statement: &influxql.SelectStatement{
+					Fields: influxql.Fields([]*influxql.Field{
+						{Expr: &influxql.VarRef{Val: "fieldKey"}},
+						{Expr: &influxql.VarRef{Val: "fieldType"}},
+					}),
+					Sources:    rewriteSources(stmt.Sources, "_fieldKeys", stmt.Database),
```

Comment:
This source hits:

```
        case "_fieldKeys":
                return NewFieldKeysIterator(s, opt)
```

It seems we don't check for `fieldKey`, `fieldType` in the fields, we just assume those - can we also add that check in `NewFieldKeysIterator` so we don't accidentally add some subtly broken rewrite?

Reply:
i.e. it seems wrong that the following queries have exactly the same results IIUC:

```
select foo from _fieldKeys
select fieldKey, fieldType from _fieldKeys
select fieldType, fieldKey from _fieldKeys // really wrong
```